### ISSUE TITLE
egg_info for package alembic produced metadata for project name unknown. Fix your #egg=alembic fragments.

### ISSUE THREAD

**Issue by tcatut** [sqlalchemy/alembic#800]
**Describe the bug**
When using PIP to install, alembic is trying to install with "UNKNOWN" as it's package name.  This causes all kinds of strange errors; and alembic does not actually install at all.  It would appear that this only happens when doing install into a fresh environment, as the error does not show up when upgrading from previous version.

I think this issue was introduced as part of the commit linked below.  Most likely because the name= parameter isn't being passed to setup() anymore.

https://github.com/sqlalchemy/alembic/commit/365283aac593698fa483b54554f9443db3fe4933#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7

**Expected behavior**
It seems like the name= keyword should not have been removed, as it's making setuptools go haywire.

**To Reproduce**
Please try to provide a [Minimal, Complete, and Verifiable](http://stackoverflow.com/help/mcve) example, with the migration script and/or the SQLAlchemy tables or models involved.
See also [Reporting Bugs](https://www.sqlalchemy.org/participate.html#bugs) on the website.

```
virtualenv ~/venv
. venv/bin/activate
pip install alembic
```

**Error**

```
Collecting alembic
  Downloading https://files.pythonhosted.org/packages/8e/07/799a76aca0acd406e3259cc6c558ca1cdadf88250953b6c8105b421a9e33/alembic-1.5.5.tar.gz (1.2MB)
    100% |████████████████████████████████| 1.2MB 386kB/s 
  Running setup.py (path:/tmp/pip-build-R3S_Iu/alembic/setup.py) egg_info for package alembic produced metadata for project name unknown. Fix your #egg=alembic fragments.
Building wheels for collected packages: unknown, unknown
  Running setup.py bdist_wheel for unknown ... done
  Stored in directory: /home/kshutt/.cache/pip/wheels/74/3f/61/7de6e3cef766d2680a5d81b1a388286e640f6a681eb589d643
  Running setup.py bdist_wheel for unknown ... error
  Complete output from command /home/kshutt/venv/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-R3S_Iu/unknown/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpeCHVpmpip-wheel- --python-tag cp27:
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
  IOError: [Errno 2] No such file or directory: '/tmp/pip-build-R3S_Iu/unknown/setup.py'
  
  ----------------------------------------
  Failed building wheel for unknown
  Running setup.py clean for unknown
  Complete output from command /home/kshutt/venv/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-R3S_Iu/unknown/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" clean --all:
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
  IOError: [Errno 2] No such file or directory: '/tmp/pip-build-R3S_Iu/unknown/setup.py'
  
  ----------------------------------------
  Failed cleaning build dir for unknown
Successfully built unknown
Failed to build unknown
Installing collected packages: unknown
Successfully installed unknown-1.5.5
You are using pip version 9.0.1, however version 21.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

**Versions.**
 - OS:  CentOS Linux release 7.9.2009 (all patches applied up to 1/27/2021)
 - Python:  2.7.5
 - Alembic:  1.5.5
 - SQLAlchemy:  1.3.23
 - Database:  mysql  Ver 14.14 Distrib 5.7.32-35, for Linux (x86_64) using  6.2
 - DBAPI:  mysqlclient 1.4.6, MySQL-python 1.2.5

**Additional context**

**Have a nice day!**

**Comment by tcatut** [comment]
We have to go all the way back to alembic<=1.5.2 to get around this issue.  After installing the old version, we are then able to use the --upgrade option to get a newer release.

HOWEVER, this only works if we did not already attempt to install the bad version.  In that case, it actually corrupts the virtualenv somehow and we have to re-initialize the virtualenv from scratch before it will stop giving the "unknown" package name.  We have even tried deleting pip cache but that didn't help either.

**Comment by zzzeek** [comment]
hey there -

based on your log output, I noticed you are on pip 9.0.1, which appears to be six years ago in 2016: https://pypi.org/project/pip/9.0.1/

I can see what's up with this key however I can't reproduce your issue with modern versions of pip.

**Comment by tcatut** [comment]
Yeah, it's RedHat so you get an old version OOTB and have to upgrade pip manually after the virtualenv is created.  I also forgot to run activate after upgrading pip, so next command can actually use the new version.

I just re-tested and was able to get it working.

**Comment by zzzeek** [comment]
OK I can reproduce it with a brand new centos 7 container which apparently installs pip 9.0.1.      but also "name" was removed as of alembic 1.5.0, not 1.5.2..   I also am observing 1.5.3 seems to be where things "break" but nothing was changed in the source code for that version, perhaps the packaging tools Im using are the issue.  not really sure.

**Comment by vpassama** [comment]
Hi, just for your information, I went into the very same issue installing alembic 1.5.5 on a debian 8, python 2.7 inside a virtualenv with pip==20.3.4.
I managed to fix the problem by updating setuptools which was by default at version 5.5.1 (Aug 11, 2014). It updated to the 44.1.1 and then everything was ok.

**Comment by armyofevilrobots** [comment]
I just wanted to post that I ran into the exact same bug on an AWS Elastic Beanstalk instance using the Python 3.6 platform. They also use pip 9.0.3 (at the time of writing this) on that platform. We've deployed on the exact same platform literally hundreds of times in he last year or so, and this is the first time it popped up.
Since we built originally our system, they've released 3.7 and 3.8 on AL2 with pip 9.0.x replaced with pipenv 2020.8.x, which should remediate the issue.
Hopefully this is helpful to anybody else scratching their heads at some inscrutable HTTP 50X errors.
